refactor(rolldown_plugin_transform): remove unused `runtime_resolve_base` option

sapphi-red opened the pull request:
`runtime_resolve_base` option is not used and probably won't be used in the future as #6177 landed.

sapphi-red commented:
* **#6226** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/6226?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/6226?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>graphite: merge</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/rolldown?ref=merge-queue-instructions-in-stack-comment&prId=6226).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

shulaoda commented:
### Merge activity

* **Sep 16, 2:20 AM UTC**: The merge label 'graphite: merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=rolldown&repo=rolldown) once it meets the requirements.
* **Sep 16, 2:20 AM UTC**: The merge label 'graphite: merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=rolldown&repo=rolldown) once it meets the requirements.
* **Sep 16, 2:22 AM UTC**: `shulaoda` added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=rolldown&repo=rolldown).
* **Sep 16, 2:30 AM UTC**: The [Graphite merge queue](https://app.graphite.dev/merges?org=rolldown&repo=rolldown) couldn't merge this PR because **it was not satisfying all requirements** (Failed CI: 'node-test (macos-latest) / Node Test', 'node-test (windows-latest) / Node Test').
* **Sep 16, 2:48 AM UTC**: `sapphi-red` added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=rolldown&repo=rolldown).
* **Sep 16, 2:48 AM UTC**: Merged by the [Graphite merge queue](https://app.graphite.dev/merges?org=rolldown&repo=rolldown).

sapphi-red committed:
refactor(rolldown_plugin_transform): remove unused `runtime_resolve_base` option (#6226)

`runtime_resolve_base` option is not used and probably won't be used in the future as #6177 landed.
